Activity_Pattern_Tracking.js

```diff
@@ -0,0 +1,219 @@
+//       console.log('-------------------- Tracking Activity Pattern --------------------');
+//       console.log(new Date().toString());
+//       setInterval(()=> { // loop untill activity period finished
+//         isOnline().then(online =>{ // check for online conectivity
+//
+//           if(online)
+//           {
+//             console.log('\tonline');
+//             var date = new Date();
+//             const index = (date.getDay()*partsPerDay)  +  (date.getHours()*partsPerHour) + Math.floor(date.getMinutes()/deltaMinutes);
+//             activity[index] =  1; // online
+//             bufferToFile('ActivityPattern.txt',activity,()=>{console.log('\tActivity Pattern Updated')});
+//             if(index==0) // resets if it was Sunday  00:00 => 00:(deltaMinutes-1)
+//             {
+//               console.log('-------------------- Reseting Activity Pattern --------------------');
+//               console.log('\tReseting Date is Sunday, Hour 00, and at any minute betwwen 00 and '+(deltaMinutes-1));
+//               activity  = Array.apply(null, Array(  Math.ceil(activityDays * partsPerDay)  ) ).map(Number.prototype.valueOf,0);
+//               bufferToFile('ActivityPattern.txt',activity,()=>{console.log('\tActivity Pattern Reseted')});
+//             }
+//           } else console.log('\toffline');
+//
+//         });
+//       }, the_interval);
+//
+//     });
+//   } else {  //  Activity Pattern File doesn't exit
+//
+//     return console.error('Error ! Activity Pattern File does not exists on path: '+activityPath);
+//     }
+// }
+
+
+
+//
+// createActivityPatternFile((activity)=>{
+//   console.log('\tActivity file was Created using the Defaults parameters');
+//   trackActivityPattern(1,1,'testing.txt');
+// },1,1,'testing.txt');
+
+// loadActivityPattern((activity)=>{
+//   console.log('activity file path was loaded using the Defaults activityPath');
+//   console.log('activity.length '+activity.length);
+//   const v= new Date();
+//   console.log(v.getDay());
+// });
+
+//console.log('Tracking  using the Defaults activityPath');
+//trackActivityPattern();
+
+// console.log('Calculating score of compatibility..');
+//
+// loadActivityPattern((activity)=>{
+//   console.log('activity pattern of host is loaded..');
+//   calculateMatching(activity, (score)=>{
+//       console.log(score);
+//   });
+// });
+
+
+console.log('Calculating score of availability..');
+
+loadActivityPattern((activity)=>{
+  calculateHostScore('10.40.60.62',activity, (score)=>{
+      console.log(score);
+  });
+});
+
+
+// you can also Track under specified deltaMinutes, activityDay, and activityPath
+
+   // trackActivityPattern(2);  //deltaMinutes=2,activityDays=7,activityPath='ActivityPattern.txt'
+
+   // trackActivityPattern(2,11); //deltaMinutes=2,activityDays=11,activityPath='ActivityPattern.txt'
+
+   // trackActivityPattern(2,11,'anyfilepath.txt');//deltaMinutes=2,activityDays=11,activityPath='anyfilepath.txt'
```

Comment:
is this code being used? If yes we should clean it up and put it under 'utils', if not then remove it